CONTRIBUTING.md

```diff
@@ -0,0 +1,200 @@
+# Contributing
+
+## Where to start
+
+All contributions, bug reports, bug fixes, documentation improvements, enhancements, and ideas are welcome.
+
+The best place to start is to check the [issues](https://github.com/rust-practice/cargo-leet)
+for something that interests you.
+
+## Bug Reports
+
+* Explain what is currently happening and what you expect instead.
+
+## Working on the code
+
+### Fork the project
+
+In order to work on the project you will need your own fork. To do this click the "Fork" button on
+this project.
+
+Once the project is forked clone it to your local machine:
+
+```sh
+git clone git@github.com:your-user-name/cargo-leet
+cd cargo-leet
+git remote add upstream git@github.com:rust-practice/cargo-leet.git
+```
+
+This creates the directory cargo-leet and connects your repository to the upstream (main project) repository.
+
+### Creating a branch
+
+You want your main branch to reflect only production-ready code, so create a feature branch for
+making your changes. For example:
+
+```sh
+git checkout -b my-new-feature
+```
+
+This changes your working directory to the my-new-feature branch. Keep any changes in this branch
+specific to one bug or feature so the purpose is clear. You can have many my-new-features and switch
+in between them using the git checkout command.
+
+When creating this branch, make sure your main branch is up to date with the latest upstream
+main version. To update your local main branch, you can do:
+
+```sh
+git checkout main
+git pull upstream main --ff-only
+```
+
+### Code linting, formatting, and tests
+
+You can run linting on your code at any time with:
+
+```sh
+cargo clippy
+```
+
+To format the code run:
+
+```sh
+cargo fmt
+```
+
+To run the tests:
+
+```sh
+cargo test
+```
+
```

Comment:
Maybe we should remove `chargo check` as both `cargo clippy` and `cargo t` will do this